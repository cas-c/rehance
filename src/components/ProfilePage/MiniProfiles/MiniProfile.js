import React from 'react';
import PropTypes from 'prop-types';

const MiniProfile = ({ user }) => (
    <div className='card mini-profile'>
        <div className='card-content'>
            <div className='media'>
                <div className='media-left'>
                    <figure className='image is-48x48'>
                        <img alt='' src={user.images['115']} />
                    </figure>
                </div>
                <div className='media-content'>
                    <p className='title is-5'>{user.display_name}</p>
                    <p className='subtitle is-6'>{user.username}</p>
                </div>
            </div>
        </div>
    </div>
);

MiniProfile.propTypes = {
    user: PropTypes.shape({
        images: PropTypes.objectOf(PropTypes.string).isRequired,
        display_name: PropTypes.string,
        username: PropTypes.string
    })
}

MiniProfile.defaultProps = {
    user: {
        images: { 115: 'placeholder image url' },
        display_name: 'Default User',
        username: 'username'
    }
}

export default MiniProfile;
