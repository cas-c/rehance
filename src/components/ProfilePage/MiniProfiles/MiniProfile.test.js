import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import MiniProfile from './MiniProfile';

const mockUser = {
    images: {
        115: 'testImageUrl'
    },
    display_name: 'Test User',
    username: 'miniprofileuser'
}

describe('MiniProfile', () => {
    it('should render correctly', () => {
        const output = shallow(<MiniProfile user={mockUser} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
})