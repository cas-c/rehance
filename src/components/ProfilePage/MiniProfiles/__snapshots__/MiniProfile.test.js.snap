// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniProfile should render correctly 1`] = `
<div
  className="card mini-profile"
>
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            alt=""
            src="testImageUrl"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <p
          className="title is-5"
        >
          Test User
        </p>
        <p
          className="subtitle is-6"
        >
          miniprofileuser
        </p>
      </div>
    </div>
  </div>
</div>
`;
